pkg/operator/osd/osd.go

```diff
@@ -269,15 +279,27 @@ func (c *Cluster) osdContainer(devices []Device, directories []Directory, select
 		envVars = append(envVars, locationEnvVar(config.Location))
 	}
 
-	privileged := true
+	privileged := false
+	// elevate to be privileged if it is going to mount devices
+	if devMountNeeded {
+		privileged = true
+	}
+	runAsUser := int64(0)
```

Comment:
Dont these three options need to be only set when `devMountNeeded` is false?

Reply:
I believe ceph only can run as root, and the root filesystem must be read/write.  I am just making it explicit so the intent is clear.   I am preparing another branch that will propose to do the same for the other pod types as well.